### ISSUE TITLE
Enable CUT option in source code viewer even if line is not selected

### ISSUE THREAD

**Issue by HeikoKlare** [vi-eclipse/Eclipse-Platform#23]
"The `CUT (Ctrl + X)` option is only enabled when at least one character or a line is selected; otherwise, it remains disabled.

In most IDEs (e.g VS Code, Intellij IDEA), the `CUT` operation works as follows:

- If character(s) or line(s) is selected then CUT operation is performed on them, otherwise
- The CUT operation is performed on entire line where the cursor is positioned.

I noticed Eclipse has `Ctrl + D` for that.

But still it make sense to support above behavior.  wdyt?

**Comment by HeikoKlare** [comment]
Thank you for that pointer. I have to admit that I was not aware of this functionality in other tools. Personally, I find it rather confusing. It also seems to be restricted to source code editors, since other text editors, including Notepad++ (being also used as a code editor) do not have that functionality.

So I am not in favor of providing that feature. But maybe there are other opinions? @fedejeanne?
In case we are unsure, we may also start a discussion in the official repositories about that.

In case we want to address this issue, I think we should try to place it in Platform UI text editor implementation, so that it applies to all text editors. The source code editor will then also benefit from it.

**Comment by fedejeanne** [comment]
I actually like the idea. I think it's something I would use every once in a while. 
Couple of things though:
* I think **cutting** would be more useful to me
* `CTRL + D` does not cut the line, it deletes it
* I think these kind of "line commands" could be generalized and one could let the user define its own: cutting, formatting, duplicating and deleting might be the tip of the iceberg, who knows what one might want to do with the content of the line, right?

**Comment by HeikoKlare** [comment]
Actually all this functionality is already provided. It is only about the default key bindings, isn't it? Deleting lines (CTRL+D) and duplicating lines (CTRL+ALT+UP/DOWN) is already bound to key combinations and cutting or copying whole lines can be bound via the key binding options.

![image](https://github.com/vi-eclipse/Eclipse-Platform/assets/755472/05614522-f706-4dec-ad40-65696279f3f4)

**Comment by fedejeanne** [comment]
You're right, thanks for pointing that out. No need to reinvent the wheel ^.^

I noticed that the combination `Ctrl` + `Alt` + `Up`/`Down` doesn't really work: the arrows are not detected while working in combination with `Ctrl` **and** `Alt` but they do in combination with only one of them. You can test it by trying to change the binding in the preferences. Does that also happen to you?

**Comment by HeikoKlare** [comment]
The key bindings work fine for me. What do you mean by changing the binding in the preferences? The default bindings should be ALT + Up/Down for moving the selected lines while CTRL + ALT + Up/Down copies the selected lines.

**Comment by fedejeanne** [comment]
> What do you mean by changing the binding in the preferences? 

I meant editing the binding in this field:

![Image](https://github.com/vi-eclipse/Eclipse-Platform/assets/2205684/150c2be8-e25b-4f0d-ae30-402863191820)

But I just tested again and it works, both the bindings and editing the bindings. The error must have been a fluke in my previous installation.

Since everything works and the shortcut **Cut Line** can be set to `Ctrl` + `X` by the user, this issue has been rendered moot. Closing.
